tyk-headless/templates/deployment-gw-repset.yaml

```diff
@@ -88,13 +88,28 @@ spec:
           - name: TYK_GW_STORAGE_DATABASE
             value: "{{ default "0" .Values.redis.storage.database }}"
           - name: TYK_GW_STORAGE_PASSWORD
-            value: "{{ .Values.redis.pass }}"
+            {{- if .Values.SecretName.enabled -}}
+            valueFrom:
+              secretKeyRef:
+                name: secrets-{{ include "tyk-headless.fullname" . }}
+                key: redisPass
+            {{- end }}
```

Comment:
This leads to a broken `env` definition since  when `SecretName` is not enabled then this env var has no value. Also I think logic is wrong here — when a user wants to provide the secret name, it's not actually possible without modifying the charts itself.

So maybe something along the lines of:
```.yaml
- name: TYK_GW_STORAGE_PASSWORD
  valueFrom:
    secretKeyRef:
      name: {{ if .Values.secrets.useSecretName }} {{ .Values.secrets.useSecretName }} {{ else }} secrets-{{ include "tyk-headless.fullname" . }} {{ end}}
      key: redisPass
```

Or something more pretty with a helper template call.

Same everywhere else.